CountOnMe/Model/CalculatorManager.swift

```diff
@@ -39,20 +43,33 @@ struct CalculatorManager {
         }
         stringResult += " " + operation + " "
         if operation == "=" {
-            stringResult = calculatorModel.getResult(rawString: stringResult)
+            let unformattedString = calculatorModel.getResult(rawString: stringResult)
+            guard let formattedString = formatString(unformattedString) else { return }
+            stringResult = formattedString
         }
     }
-    
-    mutating func clear(){
+
+    mutating func clear() {
         stringResult = ""
         calculatorModel.clear()
     }
+
+    private func formatString(_ unformattedString: String) -> String? {
+        let numberFormatter: NumberFormatter = .init()
+        numberFormatter.maximumFractionDigits = 4
+        numberFormatter.locale = .current
+        numberFormatter.decimalSeparator = "."
```

Comment:
**suggestion (blocking)**: One thing I forgot to tell you when I mentioned the usage of the `NumberFormatter` is that formatters (like `Number` or `Date`) are expensive to initialise.

You should store it in a `Constants.swift` (or anything else) file somewhere as a `static` constant so it's readily available and only instantiated once.